src/routes/raids/index.tsx

```diff
@@ -1,42 +1,144 @@
+  }
+  box-shadow: 0 1px 3px 0 rgba(0, 0, 0, 0.1), 0 1px 2px 0 rgba(0, 0, 0, 0.06);
+  padding: var(--space-4);
+`
+
+const Input = styled.input`
+  padding: 10px;
+  width: 300px;
+`
+
+interface FormElements extends HTMLFormControlsCollection {
+  dungeonInput: HTMLInputElement
+}
+interface FilterFormElements extends HTMLFormElement {
+  readonly elements: FormElements
+}
 
 const AllRaids = () => {
   const collectionData = useCollection<ViewRaidData>('raid-stats')
+  const [currentTab, setCurrentTab] = React.useState(tabs[0].value)
+  const [dungeon, setDungeon] = React.useState('')
 
   if (collectionData.state === 'success') {
     const data = collectionData.data
+    const dataInCurrentTab = data.filter((r) => r.status === currentTab)
+    const filteredData = dungeon
+      ? dataInCurrentTab.filter((r) =>
+          r.dungeon.toLowerCase().includes(dungeon),
+        )
+      : dataInCurrentTab
+
     return (
-      <>
-        <h1>Raids</h1>
-        <h2>Active</h2>
-        <ul>
-          {data
-            .filter((r) => r.status === 'active')
-            .map((s) => (
-              <li key={s.id}>
-                <Link to={`/raids/${s.id}`}>
-                  {s.title} | {s.dungeon}
-                </Link>
-              </li>
+      <Container>
+        <div>
+          <h3>Filters</h3>
+          <form
+            onSubmit={(e: React.SyntheticEvent<FilterFormElements>) => {
+              e.preventDefault()
+              setDungeon(
+                e.currentTarget.elements.dungeonInput.value.toLowerCase(),
+              )
+            }}
+          >
+            <div>
+              <label
+                htmlFor="dungeonInput"
+                className="block text-sm font-medium text-gray-700"
+              >
+                Dungeon
+              </label>
+              <div>
+                <Input
+                  type="text"
+                  id="dungeonInput"
+                  placeholder="hospitalrun/hospitalrun-frontend"
+                />
+              </div>
+            </div>
+          </form>
+        </div>
+
+        <div>
+          <h1>Raids</h1>
+          <Tabs aria-label="Tabs">
+            {tabs.map((tab: Tab) => (
+              <TabItem
+                key={tab.value}
+                isSelected={tab.value === currentTab}
+                {...(tab.value === currentTab
+                  ? { 'aria-current': 'page' }
+                  : {})}
+                onClick={() => setCurrentTab(tab.value)}
+              >
+                {tab.title}
+              </TabItem>
             ))}
-        </ul>
-        <h2>Completed</h2>
-        <ul>
-          {data
-            .filter((r) => r.status === 'completed')
-            .map((s) => (
-              <li key={s.id}>
-                <Link to={`/raids/${s.id}`}>
-                  {s.title} | {s.dungeon}
-                </Link>
-              </li>
+          </Tabs>
+
+          <Raids>
+            {filteredData.map((s) => (
```

Comment:
im not sure if everyone agrees, but im not really a fan of single letter variables.